### PR TITLE
[mlir] Add --list-passes option to mlir-opt

### PR DESCRIPTION
**Pull request by Natan-GabrielTiutiuIntel**
Currently, the only way to see the passes that were registered is by calling “mlir-opt --help”. However, for compilers with 500+ passes, the help message becomes too long and sometimes hard to understand. In this PR I add a new "--list-passes" option to mlir-opt, which can be used for printing only the registered passes, a feature that would be extremely useful.

### CODE REVIEW

**Review by joker-eph — APPROVED** [review]

**Comment by Natan-GabrielTiutiuIntel** [comment]
Thanks for approval @joker-eph! What are the remaining steps that have to be performed before merging this PR? Also, could you please help me with the workflow runs approval?

**Comment by joker-eph** [comment]
Sorry, I didn't pay attention that you're not a member of the project, I'll merge it now.

**Comment by joker-eph** [comment]
Actually will wait for CI to run first.

**Comment by Natan-GabrielTiutiuIntel** [comment]
Thank you, Mehdi! It seems that all checks passed, so it can be merged.

**Pull request merged**
